partcad/tests/unit/test_genai.py

```
@@ -0,0 +1,68 @@
+from partcad.part_factory_ai_openscad import PartFactoryAiScad
+
+
+def check_scad_syntax(scad_code):
+    # Create a temporary file to hold the SCAD code
+    with tempfile.NamedTemporaryFile(suffix=".scad", delete=True) as temp_file:
+        # Write the SCAD code to the temporary file
+        temp_file.write(scad_code.encode("utf-8"))
+        temp_file.flush()  # Ensure content is written to disk
+
+        # Run OpenSCAD in check mode
+        result = subprocess.run(
```

Ruff (0.8.2)

`subprocess` call: check for execution of untrusted input

(S603)